Fix MkDocs example engine showing unrelated macro aliases

luadebug opened the pull request:
The Python parser was incorrectly adding AUI_DECLARATIVE_FOR to the highlight tokens for ALL examples, regardless of the symbol being documented. This caused examples with AUI_WITH_STYLE to show AUI_DECLARATIVE_FOR highlighting when documenting PropertyListRecursive.

Changes:
- Added _MACRO_ALIASES map to define macro-to-symbol relationships
- Created _add_relevant_macro_aliases() helper function
- Function only adds macro aliases when:
  1. The macro appears in the snippet AND
  2. The macro relates to the tokens being documented
- Replaced all 5 hardcoded AUI_DECLARATIVE_FOR checks with the new helper
- Added support for AUI_WITH_STYLE ↔ PropertyListRecursive mapping

This ensures macro aliases are only highlighted in relevant examples, fixing the issue where unrelated macros were being highlighted.

luadebug commented:
/gemini review

Alex2772 commented:
Still provides `AUI_DECLARATIVE_FOR` for unrelated pages. Use search bar

<img width="1767" height="1710" alt="Screenshot_20251103_091055" src="https://github.com/user-attachments/assets/2fba7851-abec-42d0-8868-7ef1a62d25b7" />

luadebug committed:
fix: MkDocs example engine showing unrelated macro aliases

Add _add_relevant_macro_aliases helper that only augments highlight
tokens with relevant macro aliases when the alias actually appears
in the snippet. Add _compute_hl_lines wrapper to centralize error
handling. Remove repetitive try-except blocks throughout embed_doc
and gen_pages.

luadebug committed:
fix: address doc build crashes and search pollution

- Fix C++ parser crashes on non-matching comment lines, enum forward
  declarations, and None returns from enum parsing
- Escape C++ lambda introducer []() to prevent markdown link parsing
- Add post-build hook to strip AUI_DECLARATIVE_FOR from search index
  on pages unrelated to AForEachUI

fix: MkDocs example engine showing unrelated macro aliases

- Add _add_relevant_macro_aliases helper for selective macro alias highlighting
- Add _compute_hl_lines wrapper to centralize error handling
- Remove repetitive try-except blocks throughout embed_doc and gen_pages
- Remove hypertext links from example titles and note summaries
- Wrap source paths in backticks to prevent auto-linking

luadebug committed:
chore: add pixi task for docs build and serve

luadebug committed:
fix: prioritize name-based snippet extraction over strong patterns

luadebug committed:
fix: extend highlight lines to cover full brace-delimited blocks

luadebug committed:
fix: prevent duplicate Examples headers on macro pages

luadebug committed:
fix: always highlight matched line in additon to block range

luadebug committed:
perf: add C++ parser cache for faster rebuilds

- Cache parsed header results to .parser_cache.pkl keyed by file mtime
- Subsequent builds skip re-parsing unchanged .h files
- Enable --dirty for both build and serve tasks

luadebug committed:
fix: update Visual Studio generator version to 18 2026 in build workflows

luadebug committed:
fix: prioritize documented name over universal AUI_DECLARATIVE_FOR priority

The _example_priority function was giving universal priority 0 to any
example containing AUI_DECLARATIVE_FOR on ALL pages, drowning out
examples actually relevant to the documented macro.

luadebug committed:
fix: revert Visual Studio generator to correct version 17 2022

luadebug committed:
fix: prevent unrelated examples on macro pages

- Remove URL-based expansion in _expand_names_with_index_aliases
  (was adding ALL symbols sharing the same header file URL)
- Add name-existence check in list fallback before strong patterns
  (was matching AUI_DECLARATIVE_FOR even when documented name absent)
- Guard against missing src key in page_examples

luadebug committed:
chore: update Visual Studio generator to 18 2026

luadebug committed:
refactor: clean up examples engine per reviewer feedback

- Centralize macro alias logic in _add_relevant_macro_aliases helper
- Add _compute_hl_lines wrapper to remove repetitive try-except blocks
- Strip leading blank lines in hl_lines computation
- Extend hl_lines to cover full brace-delimited blocks
- Prioritize documented name over strong patterns in snippet extraction
- Remove URL-based index alias expansion
- Convert example title links to plain text
- Add parser cache for faster rebuilds
- Fix C++ parser edge cases (None crashes, forward enum decls)
- Escape C++ lambda []() from being parsed as markdown links
- Add search index filtering hook for AUI_DECLARATIVE_FOR
- Add port cleanup to pixi serve task

luadebug committed:
Resolve review feedback: guard snippets, fix title formatting, cache versioning, consume_doc, CSS scope, gitignore

luadebug committed:
fix: remove unused styles for plain-text example links in documentation admonitions

luadebug committed:
gitignore: add pixi.lock to stop tracking generated lockfile

luadebug committed:
fix: move class_examples init before non-class check to prevent NameError and stale data leaks

luadebug committed:
fix: normalize class_examples entries to have src field for fallback paths

luadebug committed:
fix: update Visual Studio generator from 18 2026 to 17 2022

luadebug committed:
fix: add installation step for Visual Studio 2022 Build Tools in CI workflows

luadebug committed:
fix: generate mkdocs-compatible anchor slugs from category names in example() function

luadebug committed:
fix: resolve review batch - title helper, MSVC step, enum warning, pixi platforms, non-class guard, cache fingerprint, shared helpers

luadebug committed:
fix: update build workflows to install Visual Studio 2019 Build Tools and adjust CMake flags

luadebug committed:
fix: update CI workflows to use Visual Studio 16 2019 and adjust CMake flags

luadebug committed:
fix: enhance CI workflows with MSVC setup for Windows Ninja and update cache fingerprint logic in Python generators

luadebug committed:
fix: update CI workflows to use Windows 2019 and remove Visual Studio 2019 Build Tools installation steps; refactor caching logic in cpp_parser.py and doxygen.py

luadebug committed:
fix: enhance CI workflows for Windows ARM64 Ninja and improve example priority handling in documentation generation

luadebug committed:
fix: add installation step for Visual Studio 2019 Build Tools and setup MSVC for Windows

luadebug committed:
fix: refine condition in _find_first_match to limit AForEachUI checks

luadebug committed:
fix: update build configuration to use windows-latest and enhance example snippet handling in documentation generation

luadebug committed:
Assume OpenSSL 3.6.3 is broken recipe because /MD not found for replace.

luadebug committed:
fix: update deployment workflow to use windows-latest and add installation steps for Visual Studio 2019 Build Tools; refactor example printing logic in documentation generation

luadebug committed:
fix: remove redundant Windows build configuration and improve blank line handling in example snippets

luadebug committed:
Assume problem is not openssl package - CLEAR CACHED PACKAGES

luadebug committed:
fix: remove unused imports and cache fingerprint; add handling for anonymous enums in parser

luadebug committed:
Clean cache and build again
